mesonbuild/dependencies/mpi.py

```diff
@@ -244,3 +247,53 @@ def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any],
         self.compile_args = ['-I' + incdir, '-I' + os.path.join(incdir, post)]
         if self.language == 'fortran':
             self.link_args.append('-l' + os.path.join(libdir, 'msmpifec'))
+
+
+class IMPIDependency(SystemDependency):
+
+    """Intel(R) MPI for Windows."""
+
+    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any],
+                 language: T.Optional[str] = None):
+        super().__init__(name, env, kwargs, language=language)
+        # only for windows
+        if not self.env.machines[self.for_machine].is_windows():
+            return
+        # only for x86 and x64
+        arch = detect_cpu_family(self.env.coredata.compilers.host)
+        if arch not in ('x86', 'x86_64'):
+            self.is_found = False  # TODO: is this line correct?
```

Comment:
I'm not sure when the `is_found` attribute should be set. Is this like correct, or should I just return without setting the attribute?

Reply:
It is technically true either way. The MSMPI dependency lookup sets it explicitly to False, though I am not entirely sure why. The mpicc handling will have it set to True by the inherited ConfigTool class based on trying to run the tool, so doing additional checks means resetting it back to False if the config-tool was found but the results were invalid.

I suppose you could omit it here.

Reply:
I've updated the PR removing support for 32 bits. Intel MPI have dropped support for version 4.1 (the last one supporting 32bit Windows) a long long time ago.